packages/themes/src/meteor/components/IconButton/themeOverrides.ts

```diff
@@ -2,122 +2,142 @@ import type { Components, Theme } from '@mui/material'
 import { alpha, buttonBaseClasses } from '@mui/material'
 
 export const MonorailIconButtonOverrides: Components<Theme>['MuiIconButton'] = {
-  defaultProps: {},
   styleOverrides: {
     root: ({
-      ownerState: {
-        color = 'primary',
-        variant = 'chromeless',
-        shape = 'rounded',
-      },
+      ownerState: { color = 'primary', variant = 'contained' },
       theme,
     }) => {
-      return {
-        ...(shape === 'rounded' && {
-          borderRadius: 4,
-        }),
-        ...(color !== 'inherit' && {
-          color: theme.palette[color].lowEmphasis.contrastText,
-          [`&.${buttonBaseClasses.focusVisible}`]: {
-            boxShadow: `0 0 0 3px ${theme.palette[color].focusRing.outer}`,
-            border: `1px solid ${theme.palette[color].focusRing.inner}`,
-          },
-          [`&.${buttonBaseClasses.disabled}`]: {
-            color: theme.palette[color].lowEmphasis.contrastText,
-            ...(variant === 'contained' && {
+      const roundedStyles = {
+        borderRadius: '4px',
+      }
+
+      const disabledStyles = {
+        [`&.${buttonBaseClasses.disabled}`]: {
+          opacity: theme.palette.action.disabledOpacity,
+        },
+      }
+
+      const containedStyles =
+        color !== 'inherit'
+          ? {
+              '&::before': {
+                content: '""',
+                position: 'absolute',
+                top: '1px',
+                left: '1px',
+                zIndex: 1,
+                borderRadius: theme.spacing(1),
+                width: 'calc(100% - 2px)',
+                height: 'calc(100% - 2px)',
+                backgroundColor: 'transparent',
+                transition: theme.transitions.create(['background-color'], {
+                  duration: theme.transitions.duration.shortest,
+                }),
+              },
+              color: theme.palette[color].lowEmphasis.contrastText,
```

Comment:
```suggestion
              color: theme.palette[color].contrastText,
```
<img width="332" alt="image" src="https://github.com/user-attachments/assets/55f63be0-bc79-4806-b56f-77f748690b34">
<img width="329" alt="image" src="https://github.com/user-attachments/assets/5fcd0778-c0b8-4778-8659-079a383258ec">